src/Illuminate/Auth/Events/Failed.php

```diff
@@ -19,14 +19,24 @@ class Failed
     public $credentials;
 
     /**
+     * The request.
+     *
+     * @var \Illuminate\Http\Request
+     */
+    protected $request;
+
+    /**
      * Create a new event instance.
      *
      * @param  \Illuminate\Contracts\Auth\Authenticatable|null  $user
      * @param  array  $credentials
+     * @param  \Illuminate\Http\Request  $request
+     * @return void
      */
-    public function __construct($user, $credentials)
+    public function __construct($user, $credentials, $request)
```

Comment:
Is there much benefit to maybe make this `$request = null` to reduce breaking changes? If someone was manually calling this themselves - they'll have to pass the request now...